go/felix/rules/static.go

```diff
@@ -429,6 +450,87 @@ func (r *DefaultRuleRenderer) StaticNATOutputChains(ipVersion uint8) []*Chain {
 	}}
 }
 
+func (r *DefaultRuleRenderer) StaticRawTableChains(ipVersion uint8) []*Chain {
+	return []*Chain{
+		r.failsafeInChain(),
+		r.failsafeOutChain(),
+		r.StaticRawPreroutingChain(ipVersion),
+		r.StaticRawOutputChain(),
+	}
+}
+
+func (r *DefaultRuleRenderer) StaticRawPreroutingChain(ipVersion uint8) *Chain {
+	rules := []Rule{}
+
+	// For safety, clear all our mark bits before we start.  (We could be in append mode and
+	// another process' rules could have left the mark bit set.)
+	rules = append(rules,
+		Rule{Action: ClearMarkAction{Mark: r.allMarkBits()}},
+	)
+
+	// Set a mark on the packet if it's from a workload interface.
+	for _, ifacePrefix := range r.WorkloadIfacePrefixes {
+		rules = append(rules, Rule{
+			Match:  Match().InInterface(ifacePrefix + "+"),
+			Action: SetMarkAction{Mark: r.IptablesMarkFromWorkload},
+		})
+	}
+
+	if ipVersion == 6 {
+		// Apply strict RPF check to packets from workload interfaces.  This prevents
+		// workloads from spoofing their IPs.  Note: non-privileged containers can't
+		// usually spoof but privileged containers and VMs can.
+		//
+		// We only do this for IPv6 because the IPv4 RPF check is handled via a sysctl.
+		// In addition, the IPv4 check is complicated by the fact that we have special
+		// case handling for DHCP to the host, which would require an exclusion.
+		rules = append(rules, r.DropRules(
+			Match().MarkSet(r.IptablesMarkFromWorkload).
+				RPFCheckFailed(),
+		)...)
+	}
+
+	rules = append(rules,
+		// Send non-workload traffic to the untracked policy chains.
+		Rule{Match: Match().MarkClear(r.IptablesMarkFromWorkload),
+			Action: JumpAction{Target: ChainDispatchFromHostEndpoint}},
+		// Then, if the packet was marked as allowed, accept it.  Packets also return here
+		// without the mark bit set if the interface wasn't one that we're policing.  We
+		// let those packets fall through to the user's policy.
+		Rule{Match: Match().MarkSet(r.IptablesMarkAccept),
+			Action: AcceptAction{}},
+	)
+
+	return &Chain{
+		Name:  ChainRawPrerouting,
+		Rules: rules,
+	}
+}
+
+func (r *DefaultRuleRenderer) allMarkBits() uint32 {
+	return r.IptablesMarkFromWorkload |
+		r.IptablesMarkAccept |
+		r.IptablesMarkNextTier
+}
+
+func (r *DefaultRuleRenderer) StaticRawOutputChain() *Chain {
+	return &Chain{
+		Name: ChainRawOutput,
+		Rules: []Rule{
+			// For safety, clear all our mark bits before we start.  (We could be in
+			// append mode and another process' rules could have left the mark bit set.)
+			{Action: ClearMarkAction{Mark: r.allMarkBits()}},
+			// Then, jump to the untracked policy chains.
+			{Action: JumpAction{Target: ChainDispatchToHostEndpoint}},
+			// Then, if the packet was marked as allowed, accept it.  Packets also
+			// return here without the mark bit set if the interface wasn't one that
+			// we're policing.
+			{Match: Match().MarkSet(r.IptablesMarkAccept),
+				Action: AcceptAction{}},
+		},
+	}
+}
```

Comment:
I think I can live with someone being able to do that.  Policing it at the config stage would be tricky (since the interface prefix could be changed later).  Inserting the check into the iptables chain adds latency to each packet just for that case, I think, which doesn't feel great.

There is no raw FORWARD table, the raw table operates before the routing decision so untracked policy does affect through traffic, but only at ingress.